src/internal/clusterstate/v2.5.0/collection.go

```diff
@@ -279,13 +282,19 @@ func migratePostgreSQLCollection(ctx context.Context, tx *pachsql.Tx, name strin
 
 				updateFields := []string{}
 				for k := range params {
-					updateFields = append(updateFields, fmt.Sprintf("%s = :%s", k, k))
+					if k == "proto" {
+						updateFields = append(updateFields, fmt.Sprintf("proto = decode('%v', 'hex') ", hex.EncodeToString([]byte(params["proto"].([]uint8)))))
+					} else {
+						updateFields = append(updateFields, fmt.Sprintf("%s = '%s'", k, params[k]))
+					}
 				}
-				params["oldKey"] = oldKey
 
-				query := fmt.Sprintf("update collections.%s set %s where key = :oldKey", col.table, strings.Join(updateFields, ", "))
+				query := fmt.Sprintf("update collections.%s set %s where key = '%s'", col.table, strings.Join(updateFields, ", "), oldKey)
```

Comment:
If you choose to take my structured batcher advice below, I think that you could build a query string thus:

```go
var (
  query = fmt.Sprintf(`update collections.%s set `)
  cols []string
  vals []any
)
⋮
for k, v := range params {
  cols = append(cols, k+"=?")
  vals = append(vals, v)
}
vals = append(vals, oldKey)
batcher.Add(ctx, fmt.Sprintf(`update collections.%s set %s where key = ?`, col.table, strings.Join(cols, ","), vals)
```

I think that doing it this way would free you from needing to hex encode the protos, since the bytes would be properly handled by the `sql` library.